parlai/tasks/coco_caption/agents.py

```diff
@@ -0,0 +1,301 @@
+except Exception as e:
+    raise ModuleNotFoundError('Need to install Pytorch: go to pytorch.org')
+from torch.utils.data import Dataset
+from parlai.agents.mlb_vqa.mlb_vqa import VqaDictionaryAgent
+
+import os
+import json
+
+# There is no real dialog in this task, so for the purposes of display_data, we
+# include a generic question that applies to all images.
+QUESTION = "Describe the above picture in a sentence."
+
+
+def _path(opt, version):
+    if version == '2014':
+        build_2014(opt)
+        buildImage_2014(opt)
+    elif version == '2017':
+        build_2017(opt)
+        buildImage_2017(opt)
+    else:
+        raise Exception('Unknown version for COCO Captions: %s' % version)
+
+    dt = opt['datatype'].split(':')[0]
+
+    if dt == 'train':
+        annotation_suffix = 'train{}'.format(version)
+        img_suffix = os.path.join('train{}'.format(version),
+                                  'COCO_train{}_'.format(version))
+    elif dt == 'valid':
+        annotation_suffix = 'val{}'.format(version)
+        img_suffix = os.path.join('val{}'.format(version),
+                                  'COCO_val{}_'.format(version))
+    elif dt == 'test':
+        annotation_suffix = 'None'
+        img_suffix = os.path.join('test{}'.format(version),
+                                  'COCO_test{}_'.format(version))
+    else:
+        raise RuntimeError('Not valid datatype.')
+
+    test_info_path = os.path.join(opt['datapath'],
+                                  'COCO_{}_Caption'.format(version),
+                                  'annotations',
+                                  'image_info_test{}.json'.format(version))
+
+    annotation_path = os.path.join(opt['datapath'],
+                                   'COCO_{}_Caption'.format(version),
+                                   'annotations',
+                                   'captions_' + annotation_suffix + '.json')
+
+    image_path = os.path.join(opt['datapath'], 'COCO-IMG', img_suffix)
+
+    return test_info_path, annotation_path, image_path
+
+
+class COCODataset(Dataset):
+    """A Pytorch Dataset utilizing streaming"""
+    def __init__(self, opt):
+        self.opt = opt
+        self.use_att = opt.get('attention', False)
+        self.use_hdf5 = not opt.get('no_hdf5', False)
+        self.datatype = self.opt.get('datatype')
+        self.training = self.datatype.startswith('train')
+        self.num_epochs = self.opt.get('num_epochs', 0)
+        self.image_loader = ImageLoader(opt)
+        test_info_path, annotation_path, self.image_path = _path(opt)
+        self._setup_data(test_info_path, annotation_path, opt.get('unittest', False))
+        if self.use_hdf5:
+            try:
+                import h5py
+                self.h5py = h5py
+            except ModuleNotFoundError:
+                raise ModuleNotFoundError('Need to install h5py - `pip install h5py`')
+            self._setup_image_data()
+        self.dict_agent = VqaDictionaryAgent(opt)
```

Comment:
as this dict agent is meant for VQA, we should probably use the standard `DictAgent` in dict.py.